packages/runtime-core/src/warning.ts

```
@@ -78,6 +78,66 @@ export function warn(msg: string, ...args: any[]): void {
+  // and inside the warnHandler's pre-formatted message string.
+  const suffix = `\n${suggestion}`
+
+  if (appWarnHandler) {
+    callWithErrorHandling(
+      appWarnHandler,
+      instance,
+      ErrorCodes.APP_WARN_HANDLER,
+      [
+        // eslint-disable-next-line no-restricted-syntax
+        msg +
+          args.map(a => a.toString?.() ?? JSON.stringify(a)).join('') +
```

ESLint

[error] Our output target is ES2016, and optional chaining results in verbose helpers and should be avoided.

(no-restricted-syntax)